platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/TransactionPoolNexus.java

```diff
@@ -172,22 +177,21 @@ public synchronized boolean submitApplicationTransaction(@NonNull final SwirldTr
      * @return true if successful
      */
     public synchronized boolean submitTransaction(
-            @NonNull final ConsensusTransactionImpl transaction, final boolean priority) {
+            @NonNull final OneOf<PayloadOneOfType> transaction, final boolean priority) {
 
         Objects.requireNonNull(transaction);
+        final boolean isSystem = STATE_SIGNATURE_PAYLOAD.equals(transaction.kind());
```

Comment:
create util for checking isSystem, also make `Transaction.isSystem` use this

Reply:
done